datafusion/optimizer/src/single_distinct_to_groupby.rs

```diff
@@ -280,11 +261,7 @@ impl OptimizerRule for SingleDistinctToGroupBy {
                         outer_group_exprs,
                         outer_aggr_exprs,
                     )?);
-
-                    Ok(Some(LogicalPlan::Projection(Projection::try_new(
-                        alias_expr,
-                        Arc::new(outer_aggr),
-                    )?)))
+                    Ok(Some(project(outer_aggr, alias_expr)?))
```

Comment:
The function `project()` will internally call `columnize_expr`, which can simplify the code here.